app/src/main/java/org/wikipedia/feed/onthisday/OnThisDayFragment.java

```diff
@@ -310,63 +315,78 @@ public int getItemViewType(int position) {
         }
 
-        private void setRecycler() {
-            if (pagesRecycler != null) {
-                pagesRecycler.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.HORIZONTAL, false));
-                setUpRecycler(pagesRecycler);
-            }
-        }
-
-        public void setFields(@NonNull final OnThisDay.Event event, @Nullable OnThisDay.Event prevEvent) {
-            setPagesRecycler(event);
-            setPads();
+        public void setFields(@NonNull final OnThisDay.Event event, int position) {
             descTextView.setText(event.text());
+            descTextView.setVisibility(TextUtils.isEmpty(event.text()) ? GONE : VISIBLE);
             yearTextView.setText(DateUtil.yearToStringWithEra(event.year()));
             yearsInfoTextView.setText(DateUtil.getYearDifferenceString(event.year()));
-            if (prevEvent != null && prevEvent.year() == event.year()) {
-                yearContainer.setVisibility(View.GONE);
-                yearSpace.setVisibility(View.GONE);
-            } else {
-                yearContainer.setVisibility(View.VISIBLE);
-                yearSpace.setVisibility(prevEvent == null ? View.GONE : View.VISIBLE);
-            }
-        }
-
-        private void setPads() {
-            int pad1 = (int) DimenUtil.dpToPx(PADDING1);
-            int pad2 = (int) DimenUtil.dpToPx(PADDING2);
-            int pad3 = (int) DimenUtil.dpToPx(PADDING3);
-
-            descTextView.setPaddingRelative(pad1, 0, 0, 0);
-            pagesRecycler.setPaddingRelative(pad2, 0, 0, 0);
-            yearTextView.setPaddingRelative(pad3, 0, 0, 0);
+            setPagesViewPager(event);
         }
 
-        private void setPagesRecycler(OnThisDay.Event event) {
+        private void setPagesViewPager(OnThisDay.Event event) {
             if (event.pages() != null) {
-                OnThisDayCardView.RecyclerAdapter recyclerAdapter = new OnThisDayCardView.RecyclerAdapter(getChildFragmentManager(), event.pages(), wiki, false);
-                pagesRecycler.setAdapter(recyclerAdapter);
+                ViewPagerAdapter viewPagerAdapter = new ViewPagerAdapter(getChildFragmentManager(), event.pages(), wiki);
+                pagesViewPager.setAdapter(viewPagerAdapter);
+                pagesViewPager.setOffscreenPageLimit(2);
+                 new TabLayoutMediator(pagesIndicator, pagesViewPager, (tab, position) -> { }).attach();
+                pagesViewPager.setVisibility(VISIBLE);
+                pagesIndicator.setVisibility(event.pages().size() == 1 ? GONE : VISIBLE);
             } else {
-                pagesRecycler.setVisibility(View.GONE);
+                pagesViewPager.setVisibility(GONE);
+                pagesIndicator.setVisibility(GONE);
             }
         }
+
+        public void animateRadioButton() {
+            Animation pulse = AnimationUtils.loadAnimation(getContext(), R.anim.pulse);
+            radioButtonImageView.startAnimation(pulse);
+        }
```

Comment:
Not sure but would it be possible to not show the animation for 2+ dots at a time?

Reply:
We tried a few different ways and Robin okayed this animation @cooltey .. when i tried to show only one by using 'firstVisibleItem' if they scrolled even a little fast they would miss the animation.